hazelcast/src/main/java/com/hazelcast/jet/impl/connector/ReadJdbcP.java

```diff
@@ -145,21 +146,63 @@ public static <T> ProcessorMetaSupplier supplier(
             ToResultSetFunction resultSetFn,
             FunctionEx<? super ResultSet, ? extends T> mapOutputFn) {
 
-        return ProcessorMetaSupplier.preferLocalParallelismOne(ConnectorPermission.jdbc(null, ACTION_READ),
-                SecuredFunctions.readJdbcProcessorFn(dataConnectionRef.getName(), resultSetFn, mapOutputFn));
+        return ProcessorMetaSupplier.preferLocalParallelismOne(
+                new ProcessorSupplier() {
+
+                    private transient JdbcDataConnection dataConnection;
+
+                    @Override
+                    public void init(@Nonnull Context context) {
+                        dataConnection = context.dataConnectionService()
+                                .getAndRetainDataConnection(dataConnectionRef.getName(), JdbcDataConnection.class);
+                    }
+
+                    @Nonnull
+                    @Override
+                    public Collection<? extends Processor> get(int count) {
+                        return IntStream.range(0, count)
+                                .mapToObj(i -> new ReadJdbcP<T>(() -> dataConnection.getConnection(), resultSetFn, mapOutputFn))
+                                .collect(Collectors.toList());
+                    }
+
+                    @Override
+                    public void close(@Nullable Throwable error) {
+                        if (dataConnection != null) {
+                            dataConnection.release();
+                        }
+                    }
+                });
+    }
+
+    private static <T> ProcessorSupplier readJdbcProcessorFn(
+            FunctionEx<ProcessorSupplier.Context, ? extends Connection> newConnectionFn,
+            ToResultSetFunction resultSetFn,
+            FunctionEx<? super ResultSet, ? extends T> mapOutputFn
+    ) {
+        return new ProcessorSupplier() {
+
+            private transient Context context;
```

Comment:
IMHO doesn't have to be transient, as it won't be ever sent via network

Reply:
I think it is because ProcessorSupplier is serializable and the field is made transient to avoid warnings from compiler